### PR TITLE
Better JavaScript property validation

### PR DESCRIPTION
**Pull request by externl**
Part of #2746 

I also did cleanup/fixes to the other languages as JS was the final language

### CODE REVIEW

File: config/makeprops.py
Comment (externl) [inline config/makeprops.py:1]: I reworked a lot of this file. 
- cleanup now that all languages use the same setup
- removed global property classes
- changed the owner ship of prefixOnly
- rename propertyClass to propertyArray

**Review by bernardnormier — APPROVED** [review]

**Review by pepone — APPROVED** [review]
Looks good, but see comments.

We need to add a test, I added one for C# in https://github.com/zeroc-ice/ice/tree/main/csharp/test/Ice/unknownProperties but seems we don't have any test for other languages. (for a follow up PR)

File: js/src/Ice/PropertyNames.js [hunk 1]
```diff
@@ -4,64 +4,84 @@
...
+    ["IceBox", null],
+    ["IceBoxAdmin", null],
+    ["IceBridge", null],
+    ["IceGridAdmin", null],
+    ["IceGrid", null],
+    ["IceSSL", null],
+    ["IceStorm", null],
+    ["IceStormAdmin", null],
+    ["IceBT", null],
+    ["Glacier2", null],
```
[the remark this answers is not in the document]
Comment (externl) [reply js/src/Ice/PropertyNames.js:86]: Yes, these are the reserved property prefixes. We still want to issue a warning if a user is trying to use them in JavaScript.

File: js/src/Ice/LocalExceptions.js [hunk 2]
```diff
@@ -362,3 +362,13 @@ export class TwowayOnlyException extends LocalException {
         this.operation = operation;
     }
 }
+
+export class UnknownPropertyException extends RequestFailedException {
```
[the remark this answers is not in the document]
Comment (externl) [reply js/src/Ice/LocalExceptions.js]: Fixed.

**Review by pepone — APPROVED** [review]

**Pull request merged**